pkg/http/http.go

```diff
@@ -113,28 +124,52 @@ func (s *Server) WithTracing(provider trace.TracerProvider) *Server {
 	return s
 }
 
-// WithPrometheus adds a /metrics endpoint and instrument subsequently enabled groups with general http-level metrics.
-func (s *Server) WithPrometheus() *Server {
-	// Add MW with .With instead of .Use, as .Use does not allow registering MWs after routes.
-	s.router = s.router.With(PrometheusMiddleware)
-	s.router.Handle("/metrics", promhttp.Handler())
-
+// WithInsecureTraceContext configures the server to trust parent trace IDs arriving to public-facing endpoints.
+// This will make traces generated by QuickPizza to be children of client-controlled trace IDs.
+func (s *Server) WithInsecureTraceContext(insecure bool) *Server {
+	s.insecureTracing = insecure
 	return s
 }
 
+// isPublic returns whether an endpoint should be configured public for purposes of trace linking, by checking for the
+// presence of a secret token in the request.
+// It will always return true if the server was configured WithInsecureTraceContext.
+func (s *Server) isPublic(r *http.Request) bool {
+	return s.insecureTracing || r.Header.Get("X-Internal-Token") != "secret"
+}
```

Comment:
I don't get this logic. To me, the `insecureTracing` field implies that we **don't** add the link association using the "public endpoint" logic. However, if this is set to true, all traced endpoints now become "public".

Also, what's the use case for the `X-Internal-Token` header? Do we have demo flows that require this behavior? If not, again, perhaps it's not needed.

Reply:
🤔 Good catch, it is likely that I've swapped the logic here.

`X-Internal-Token` is the demo-scoped way of identifying requests originating from inside the system. In real environments an API gateway would be playing this role.